### PR TITLE
addTags

### PR DESCRIPTION
**Pull request by asenousy**
*Description of changes:* Get all resources tagged  (for metrics/cost/finding resources)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/2-landing-page/cdk/lib/landing-page-stack.ts [hunk 1]
```diff
@@ -1,46 +1,52 @@
...
+  CloudFrontWebDistribution,
+  OriginAccessIdentity
+} from '@aws-cdk/aws-cloudfront';
+import { Construct, Stack, StackProps, CfnOutput, Tags } from '@aws-cdk/core';
 
 export class LandingPageStack extends Stack {
   constructor(scope: Construct, id: string, props?: StackProps) {
     super(scope, id, props);
 
+    Tags.of(this).add('StackType', this.node.tryGetContext('service_name'));
```
Comment (flochaz) [inline source/2-landing-page/cdk/lib/landing-page-stack.ts:13]: I think the key should be ServiceName not StackType

File: source/3-landing-page-cicd/cdk/cdk.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
     "@aws-cdk/core:stackRelativeExports": "true",
     "github_alias": "<YOUR GITHUB ALIAS>",
     "github_repo_name": "aws-bootstrap-kit-examples",
-    "github_repo_branch": "main"
+    "github_repo_branch": "main",
+    "service_name": "Landing-Page-CICD"
```
Comment (flochaz) [inline source/3-landing-page-cicd/cdk/cdk.json:11]: I think the service_name should still be Landing-Page . CI/CD is a just deployment process

File: source/3-landing-page-cicd/cdk/lib/landing-page-stack.ts [hunk 3]
```diff
@@ -1,47 +1,53 @@
...
+  CloudFrontWebDistribution,
+  OriginAccessIdentity
+} from '@aws-cdk/aws-cloudfront';
+import { Construct, Stack, StackProps, CfnOutput, Tags } from '@aws-cdk/core';
 
 export class LandingPageStack extends Stack {
   constructor(scope: Construct, id: string, props?: StackProps) {
     super(scope, id, props);
 
+    Tags.of(this).add('StackType', this.node.tryGetContext('service_name'));
```
Comment (flochaz) [inline source/3-landing-page-cicd/cdk/lib/landing-page-stack.ts:13]: I think the key should be ServiceName not StackType

**Review by flochaz — CHANGES_REQUESTED** [review]

File: source/2-landing-page/cdk/lib/landing-page-stack.ts (hunk 1 above)
Comment (ijemmy) [reply source/2-landing-page/cdk/lib/landing-page-stack.ts:13]: Also need unit test

(Love how simple it is to tag all resources in CDK!)

File: source/3-landing-page-cicd/cdk/test/landing-page.test.ts [hunk 4]
```diff
@@ -1,13 +1,32 @@
...
+    context: {
+      service_name: 'Landing-Page-CICD'
+    }
+  });
+  const stack = new LandingPage.LandingPageStack(app, 'MyTestStack');
+  expectCDK(stack).to(
+    haveResource('AWS::S3::Bucket', {
+      Tags: [
+        {
+          Key: 'StackType',
```
Comment (flochaz) [inline source/3-landing-page-cicd/cdk/test/landing-page.test.ts:16]: Test can't pass since now it's ServiceName ...

**Review by flochaz — CHANGES_REQUESTED** [review]

**Comment by flochaz** [comment]
Make sure test pass

File: source/3-landing-page-cicd/cdk/test/landing-page.test.ts (hunk 4 above)
Comment (asenousy) [reply source/3-landing-page-cicd/cdk/test/landing-page.test.ts:16]: oops, sorry :) will fix it now

**Review by flochaz — APPROVED** [review]

**Pull request merged**